scripts/globals/moblin_maze_mongers/mmm_shop.lua

```diff
@@ -0,0 +1,78 @@
+-----------------------------------
+-- Moblin Maze Monglers: Shop
+-----------------------------------
+require('scripts/globals/npc_util')
+-----------------------------------
+xi = xi or {}
+xi.mmm = xi.mmm or {}
+-----------------------------------
+
+-- Maze voucher (1) menu.   -->  262
+-- Maze voucher (100) menu. -->  518
+-- Maze rune (100) menu.    -->  258
+-- Maze rune (200) menu.    -->  514
+-- Maze rune (300) menu.    -->  770
+-- Maze rune (400) menu.    --> 1026
+-- Maze rune (500) menu.    --> 1282
+-- "Take my gil" menu.      -->    7
+
+local optionTable =
+{
+    [35009] = { xi.item.MAZE_VOUCHER_02, 1, 100, 'moblin_marble' },
+    [ 2241] = { xi.item.MAZE_VOUCHER_09, 1, 100, 'moblin_marble' },
+}
+
+xi.mmm.shopOnTrigger = function(player)
+    if player:hasKeyItem(xi.ki.TATTERED_MAZE_MONGER_POUCH) then
```

Comment:
If you want to do this in chunks and merge into base, introduce a new setting and have it off by default - then you can drip in work as you want/need to, like I've done with Chocobo Raising

Reply:
I created a setting. All this content is gated behind a KI, so I limited the setting to affect getting it.